docs/source/federated-types/federated-directives.mdx

```diff
@@ -168,6 +168,18 @@ The default value is `true`.
 </tbody>
 </table>
 
+### `@interfaceObject`
+
+> **⚠️ This directive is available in Apollo Federation v2.3 and later.**
+
+```graphql
+directive @interfaceObject on OBJECT
+```
+
+Indicates that an object definition serves as an extension of another subgraph's **entity interface**. This enables a subgraph to polymorphically contribute fields to _all_ entities that implement a particular entity interface.
```

Comment:
I will say that Sylvain really does not like the word _extends_ for `@interfaceObject`, as he thinks of it more as an abstraction of the actual interface, rather than extending it. It's tricky, because I think "extends" is a quicker teaching moment to users, but it's not accurate in this case. 

For the definition in this case, I would say: "Indicates that an object definition serves as a local abstraction of another subgraph's **entity interface**. During composition, object types with the `@interfaceObject` annotation and the interface with the corresponding `@key` will be merged. This enables a subgraph to polymorphically contribute fields to _all_ entities that implement a particular entity interface. Additionally, all fields contributed by types marked by `@interfaceObject` will be added to the definition of each type implementing the entity interface in the supergraph API." 

Feel free to ignore or change as needed!